torch_xla/csrc/tensor.cpp

```diff
@@ -1649,12 +1662,13 @@ std::shared_ptr<XLATensor::Async> XLATensor::SyncTensorsGraphInternal(
       "SyncTensorsGraphInternal", tensorflow::profiler::TraceMeLevel::kInfo);
   SyncTensorCollection coll = CollectSyncTensors(*tensors, config);
   if (coll.indices.empty()) {
+    TensorCollectionBarrier(&coll);
     return nullptr;
   }
+  PostOrderData po_data = RunPostOrder(*tensors, &coll);
   DebugUtil::SaveTensorsGraphInfo("ScheduleSyncTensorsGraph", *tensors,
                                   &coll.indices);
 
-  PostOrderData po_data = RunPostOrder(*tensors, coll.indices);
```

Comment:
why do we move this line?

Reply:
This line leads to a seg fault in cases where the conditional lock needed to be called. I moved it after the `RunPostOrder`. Do you have any concerns to share here?

Reply:
Can you verify `XLA_SAVE_TENSORS_FILE` still works after you switch the order?

Reply:
+ I tested `XLA_SAVE_TENSORS_FILE` against `test_operations.py`. All tests successfully pass, and the saved file contains several HLO graphs corresponding to `ScheduleSyncTensorsGraph`.
+ However, combining this flag with the OpByOp flags fails; this is a known [issue](https://github.com/pytorch/xla/issues/3485) however.

```
XLA_GET_TENSORS_OPBYOP=1 XLA_SYNC_TENSORS_OPBYOP=1 XLA_SAVE_TENSORS_FILE=/tmp/graphs.hlo XLA_SAVE_TENSORS_FMT=hlo python test_operations.py
```